### DISCUSSION TITLE
MS Teams Integration

### DISCUSSION THREAD

**Discussion by Saketh24893** [cypress-io/cypress#26247] [Questions and Help]
Hi Team,

I have tried the plugin cypress-msteams-reporter for publishing some sample message on the channel created in Teams. But I am unable to send any message to the teams channel. Even I have configured the MS_TEAMS_WEBHOOK_URL in cypress.config.js file under env but still I am getting the error as 
![image](https://user-images.githubusercontent.com/129068193/228184368-aadd6907-bfce-4e54-9591-db532e49d1b4.png)

And this is my cypress.config.js file where i have added the env as MS_TEAMS_WEBHOOK_URL.
![image](https://user-images.githubusercontent.com/129068193/228184586-8f8d784e-de13-44f2-ab3e-42ae3ddbf405.png)

Please suggest where / what i am doing wrong and help me to resolve the problem.

Thanks in advance for your support.

**Comment by portyu9** [comment]
Hi @Saketh24893 

The immediate problem in your configuration is that `cypress-msteams-reporter` expects `MS_TEAMS_WEBHOOK_URL` as a **process environment variable**. Putting it under Cypress's `env` configuration does not make it available as:

```js
process.env.MS_TEAMS_WEBHOOK_URL
```

That explains the reporter error:

```text
Incoming webhook URL is required
```

Historically you would have supplied it from the shell/CI environment, for example:

```bash
MS_TEAMS_WEBHOOK_URL="..." npx cypress-msteams-reporter
```

rather than:

```js
e2e: {
  env: {
    MS_TEAMS_WEBHOOK_URL: '...'
  }
}
```

There is now an additional issue: **Microsoft retired the old Office 365 Connector/Incoming Webhook infrastructure in May 2026**. New integrations should use a Teams **Workflow / Power Automate webhook** instead of the old `webhook.office.com/.../IncomingWebhook/...` URL and a reporter built around that connector.

So for a current implementation I would:

1. Create a Teams Workflow using **When a Teams webhook request is received**.
2. Store that webhook URL as a CI secret.
3. Send the Cypress result to that workflow from your CI/Node reporting step.

One more important point: the screenshot contains the full webhook URL. Treat that URL as a credential and **rotate/revoke it** if it was ever publicly posted.

Please mark this answer as accepted if it helped, thank you.
